Build works now on NixOS 18.03.

I get this error when executing:

```
[davidak@X230:~]$ /nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/bin/i3pystatus
Traceback (most recent call last):
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/bin/.i3pystatus-wrapped", line 12, in <module>
    sys.exit(main())
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/lib/python3.6/site-packages/i3pystatus/__init__.py", line 49, in main
    clock_example()
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/lib/python3.6/site-packages/i3pystatus/__init__.py", line 34, in clock_example
    status.register(Clock())
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/lib/python3.6/site-packages/i3pystatus/core/modules.py", line 66, in __init__
    super(Module, self).__init__(*args, **kwargs)
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/lib/python3.6/site-packages/i3pystatus/core/settings.py", line 109, in __init__
    self.init()
  File "/nix/store/8wgq0f4f3rs8n338bl99l324pqjyjvkd-i3pystatus-2018-04-11-unstable/lib/python3.6/site-packages/i3pystatus/clock.py", line 83, in init
    locale.setlocale(locale.LC_TIME, lang)
  File "/nix/store/96wn2gz3mwi71gwcrvpfg39bsymd7gqx-python3-3.6.5/lib/python3.6/locale.py", line 598, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting

```

--- a/pkgs/applications/window-managers/i3/pystatus.nix
+++ b/pkgs/applications/window-managers/i3/pystatus.nix
@@ -20,7 +20,8 @@ python3Packages.buildPythonApplication rec {
 
   libpulseaudioPath = stdenv.lib.makeLibraryPath [ libpulseaudio ];
   ldWrapperSuffix = "--suffix LD_LIBRARY_PATH : \"${libpulseaudioPath}\"";
-  makeWrapperArgs = [ ldWrapperSuffix ]; # libpulseaudio.so is loaded manually
+  # LC_TIME != C results in locale.Error: unsupported locale setting
+  makeWrapperArgs = [ "--set LC_TIME C" ldWrapperSuffix ]; # libpulseaudio.so is loaded manually
 
   postInstall = ''
     makeWrapper ${python3Packages.python.interpreter} $out/bin/${pname}-python-interpreter \

--- a/pkgs/applications/window-managers/i3/pystatus.nix
+++ b/pkgs/applications/window-managers/i3/pystatus.nix
@@ -1,14 +1,18 @@
-{ stdenv, fetchurl, libpulseaudio, python3Packages, extraLibs ? [] }:
+{ stdenv, fetchFromGitHub, libpulseaudio, python3Packages, extraLibs ? [] }:
 
 python3Packages.buildPythonApplication rec {
-  name = "${pname}-${version}";
-  version = "3.35";
+  # i3pystatus moved to rolling release:
+  # https://github.com/enkore/i3pystatus/issues/584
+  version = "unstable-2018-04-11";
   pname = "i3pystatus";
   disabled = !python3Packages.isPy3k;
 
-  src = fetchurl {
-    url = "mirror://pypi/i/${pname}/${name}.tar.gz";
-    sha256 = "0g5m05rbqvq1qrspm6fyzky9xfhaz5pvc4hfzgdxrzijn8nfc860";
+  src = fetchFromGitHub
+  {
+    owner = "enkore";
+    repo = "i3pystatus";
+    rev = "3efbd56bb7a851f16173ec6f0eef472b6e96c7cc";
+    sha256 = "0r4mc23chxlaym7jcjnflw7mn5nbw3q8q4ix0nim7lh98yfndd3b";
   };
 
   propagatedBuildInputs = with python3Packages; [ keyring colour netifaces praw psutil basiciw ] ++

--- a/pkgs/development/python-modules/praw/default.nix
+++ b/pkgs/development/python-modules/praw/default.nix
@@ -14,6 +14,11 @@ buildPythonPackage rec {
     sha256 = "13vbh2r952ai2m6sc79psfwaj5fc8cssdg2pqpizg2mwd0l1s6lb";
   };
 
+  postPatch = ''
+    # drop upper bound of prawcore requirement
+    sed -ri "s/'(prawcore >=.+), <.+'/'\1'/" setup.py
+  '';
+
   propagatedBuildInputs = [
     requests
     decorator

